### PR TITLE
Updated usage of ApplicationStartInfo to BuildInfo

### PR DESCRIPTION
**Pull request by dhruv-vora**
**Description:** 

- Changed usage of ApplicationStartInfo to BuildInfo
- Changed ExeName to Command

This is in continuation to [open-telemetry/opentelemetry-collector/pull/3056](https://github.com/open-telemetry/opentelemetry-collector/pull/3056) and [open-telemetry/opentelemetry-collector/pull/3057](https://github.com/open-telemetry/opentelemetry-collector/pull/3057)

### CODE REVIEW

**Review by Aneurysm9 — APPROVED** [review]

File: cmd/otelcontribcol/main.go [hunk 1]
```diff
@@ -33,13 +33,14 @@ func main() {
...
 
-	info := component.ApplicationStartInfo{
-		ExeName: "otelcontribcol",
-		Version: version.Version,
-		GitHash: version.GitHash,
+	info := component.BinaryInfo{
+		Command:     "otelcontribcol",
+		Description: "OpenTelemetry Collector Contrib",
+		Version:     version.Version,
+		GitHash:     version.GitHash,
```
Comment (bogdandrutu) [inline cmd/otelcontribcol/main.go:40]: I think this was removed, you probably need to rebase

**Review by mxiamxia — APPROVED** [review]
LGTM after addressing the comments.

File: cmd/otelcontribcol/main.go (hunk 1 above)
Comment (dhruv-vora) [reply cmd/otelcontribcol/main.go]: Done and updated the PR

**Comment by bogdandrutu** [comment]
please run `make update-otel`

**Comment by bogdandrutu** [comment]
Please fix lint errors (to run locally `make golint`)

```
newrelic.go:41: File is not `gofmt`-ed with `-s` (gofmt)
	buildInfo     *component.BuildInfo
make[1]: *** [../../Makefile.Common:93: lint] Error 1
```

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**